### PR TITLE
[IMP] account: stronger constraint on repartition lines

### PR DESCRIPTION
**Pull request by oco-odoo**
We now force tax repartition lines to use the same percentages for invoices and credit notes. Thanks to that, we are able to simplify the way manual modifications made on vendor bills' account.invoice.tax entries must affect the ones of the refunds created for them. This also prepares the code for the incoming big accounting revamping.

### CODE REVIEW

**Comment by oco-odoo** [comment]
@smetl

**Comment by smetl** [comment]
@robodoo r+

**Comment by smetl** [comment]
@robodoo retry

**Comment by smetl** [comment]
@robodoo r+

**Comment by smetl** [comment]
@robodoo retry

**Comment by smetl** [comment]
@robodoo retry You can do it!

**Comment by smetl** [comment]
@d-fence :policeman: S.O.S runbot on :fire:

**Comment by d-fence** [comment]
I'm sorry, it seems that there is a forward port in progress

File: addons/account/models/account_invoice.py [hunk 1]
```diff
@@ -1693,6 +1635,35 @@ def _prepare_refund(self, invoice, date_invoice=None, date=None, description=Non
             values['date'] = date
         if description:
             values['name'] = description
+
+        # Treat refund tax lines. They should match the incoice's, except for the data
```
Comment (qdp-odoo) [inline addons/account/models/account_invoice.py]: what's an incoice?
review comment: what we do is copy data from invoice move from refund except the account_id and the tag_ids

File: addons/account/models/account_invoice.py [hunk 2]
```diff
@@ -1693,6 +1635,35 @@ def _prepare_refund(self, invoice, date_invoice=None, date=None, description=Non
...
+        # directly coming from repartition_lines, which should be replaced by their
+        # refund equivalent
+        tax_rep_ln_mapping = {}
+        for invoice_tax_entry in invoice.tax_line_ids:
+            if not invoice_tax_entry.tax_repartition_line_id.tax_id in tax_rep_ln_mapping:
```
Comment (qdp-odoo) [inline addons/account/models/account_invoice.py]: tax_id as key ? something is escaping my understanding

Comment (oco-odoo) [reply addons/account/models/account_invoice.py]: Oops. Indeed, it worked by chance (and did useless operations).

File: addons/account/models/account_invoice.py (hunk 1 above)
Comment (oco-odoo) [reply addons/account/models/account_invoice.py]: It's a small bean found on some island on the south of Asia. It's very tasty, and used in a lot of traditional meals. Never heard of incoice noodles?

**Comment by qdp-odoo** [comment]
@robodoo r+

**Pull request closed without merging**